### PR TITLE
Fix cache controller to enforce consistent JSON serialization

### PR DESCRIPTION
**Pull request by marcelveldt**
Queue items were losing their `media_item` after restarts because the in-memory cache stored full Python objects, but the database serialized them to JSON — and nested model fields didn't survive the round-trip. This caused hundreds of "Skipping queue item ... restored from cache without media_item" warnings on every startup.

- **Removed the in-memory cache layer** — the SQLite database (WAL mode, mmap, 64MB page cache) is fast enough for all paths. This eliminates the inconsistency where memory cache returned Python objects while the DB cache returned deserialized dicts.
- **Cache now always serializes through JSON on write** — non-serializable data (model objects, custom classes) raises immediately instead of silently losing data on restart.
- **Fixed all callers** to explicitly use `.to_dict()` on set and `.from_dict()` on get, removing `isinstance(dict)` workarounds in bandcamp, filesystem_local, and player_queues.
- **Removed the "restored from cache without media_item" workaround** in player_queues — this was a symptom of the serialization inconsistency, not a valid edge case.
- Added `SerializableType` alias to document the cache contract.
- Added 29 tests for the cache controller.

### CODE REVIEW

File: music_assistant/controllers/cache.py [hunk 1]
```diff
@@ -44,17 +43,22 @@
 
 BYPASS_CACHE: ContextVar[bool] = ContextVar("BYPASS_CACHE", default=False)
 
+# Type alias for data that can be stored in the cache.
+# Only JSON-serializable types are accepted - storing model objects directly
+# is not allowed as the cache always serializes/deserializes through JSON.
+# Note: tuples are not valid because JSON round-trips convert them to lists.
+SerializableType = str | int | float | bool | None | list[Any] | dict[str, Any]
```
[the remark this answers is not in the document]
Comment (marcelveldt) [reply music_assistant/controllers/cache.py]: Fixed. The `json_dumps` fallback handler (`get_serializable_value`) does handle objects with `.to_dict()` and dataclasses, which is how the `use_cache` decorator works — that's intentional since the decorator uses type annotations for reconstruction on the way out. The `SerializableType` type hint on `set()` serves as documentation for direct callers. Tuples are indeed coerced to lists by JSON, which is why the type alias excludes them.

File: music_assistant/controllers/cache/README.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+This package provides a centralized caching layer backed by SQLite. All data stored in the cache goes through JSON serialization, ensuring consistent behavior regardless of when or how the data is retrieved.
+
+## Responsibilities
+
+- Store and retrieve JSON-serializable data with key/provider/category namespacing.
+- Enforce data integrity: only `SerializableType` values (str, int, float, bool, None, list, dict) are accepted. Non-serializable objects raise `TypeError` immediately on write.
+- Support expiration, checksums, and persistent entries that survive cache clears.
+- Provide a `base_class` parameter on `get()` to automatically reconstruct model objects from cached dicts using `from_dict()`.
+- Provide a `use_cache` decorator for transparently caching provider/controller method results with automatic serialization and deserialization based on type annotations.
+- Run scheduled cleanup of expired entries.
```
[the remark this answers is not in the document]
Comment (marcelveldt) [reply music_assistant/controllers/cache/README.md:12]: Good catch — fixed in 598e0a6. Updated BBC Sounds (`Segment` dataclass: serialize with `dataclasses.asdict()`, reconstruct with `Segment(**dict)`) and OpenSubsonic (`PodcastChannel`: use `.to_dict()`/`base_class` on the cache calls).
